lib/datasource.js

```diff
@@ -468,23 +468,21 @@ DataSource.prototype.defineRelations = function(modelClass, relations) {
   // Set up the relations
   if (relations) {
     Object.keys(relations).forEach(function(rn) {
+      var targetModel;
       var r = relations[rn];
+
       assert(DataSource.relationTypes.indexOf(r.type) !== -1, 'Invalid relation type: ' + r.type);
       assert(isValidRelationName(rn), 'Invalid relation name: ' + rn);
 
-      var targetModel, polymorphicName;
-
       if (r.polymorphic && r.type !== 'belongsTo' && !r.model) {
-        throw new Error(g.f('No model specified for {{polymorphic}} %s: %s', r.type, rn));
+        throw new Error(g.f('Param "model" not specified for {{polymorphic}} %s relation: %s', r.type, rn));
       }
-
```

Comment:
regarding the whole removed `if(r.polymorphic) { .. }` block below:
- a general comment is that there shouldn't be any semantic difference between defining the relation with json definition (using `datasource.defineRelations()`) and using relations methods such as `model.hasMany(otherModel, {..})`

The block should not be here in the first place, and any definition manipulation should be handled in the relations methods